gax-grpc/src/main/java/com/google/api/gax/grpc/OperationPollingCallable.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.core.ApiClock;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.retrying.ExponentialRetryAlgorithm;
+import com.google.api.gax.retrying.ResultRetryAlgorithm;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.retrying.TimedAttemptSettings;
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.Operation;
+import com.google.longrunning.OperationsClient;
+import io.grpc.Status;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * A future callable, which relies on callback chaining to execute server polling (asking for a
+ * status of a specific operation: in progress, completed, canceled etc.). This implementation is
+ * called from {@link AttemptCallable}, essentially using for polling the same logic which is used
+ * for retrying.
+ *
+ * @param <RequestT> type of the request
+ */
+class OperationPollingCallable<RequestT> implements FutureCallable<RequestT, Operation> {
```

Comment:
So it's only when the components are combined does this create a polling algorithm. The callable class here still seems like an `OperationCheckingCallable` to me. I'm thinking of when I look at the stack trace and I see the callables in it: it would be misleading to see this callable have "polling" in the name when it doesn't do that. 

As for the inner classes: it could be a little difficult to find them when looking at other code, because they are used in an unqualified way (e.g. from `OperationCallable`). I think it would be better for them to be in the same "area" as the other retry algorithms, which is top-level classes. (Both times I have done this review, it took me extra work to find them.)

I see what you're getting at with respect to the combination of the classes together creating a polling algorithm, but you still need other components to stitch it together to create the polling; these classes here aren't sufficient to do polling in isolation.